### ISSUE TITLE
Failed to build for ARM64 / aarch64: caffe2 cmake error in CPUType.cpp.o

### ISSUE THREAD

**Issue by KumaTea** [pytorch/pytorch#49078]
## ❓ Questions and Help

I am trying to build `pytorch` for `aarch64`. I’ve tried several times, but always failed on the same error:

```
[ 73%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/CPUType.cpp.o
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/build.make:3717: caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/CPUType.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2646: caffe2/CMakeFiles/torch_cpu.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
Traceback (most recent call last):
  File "setup.py", line 724, in <module>
    build_deps()
  File "setup.py", line 312, in build_deps
    build_caffe2(version=version,
  File "/home/kuma/build/pytorch/tools/build_pytorch_libs.py", line 62, in build_caffe2
    cmake.build(my_env)
  File "/home/kuma/build/pytorch/tools/setup_helpers/cmake.py", line 346, in build
    self.run(build_args, my_env)
  File "/home/kuma/build/pytorch/tools/setup_helpers/cmake.py", line 141, in run
    check_call(command, cwd=self.build_dir, env=env)
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'install', '--config', 'Release', '--', '-j', '1']' returned non-zero exit status 2.
```

Environment:
> SoC: [Allwinner H6](https://linux-sunxi.org/H6) (quad-core A53)
> OS: Ubuntu 20.04.1 focal LTS
> Python: 3.8.5

Build options:
```
git clone https://github.com/pytorch/pytorch
cd pytorch
git pull
git submodule sync
git submodule update --init --recursive

export MAX_JOBS=1

python3.8 setup.py build
```

---

I’ve noticed that [the CI has successfully built it](https://status.openlabtesting.org/builds/builds?project=pytorch%2Fpytorch&job_name=pytorch-arm64-build-daily-master-py38), and the `whl` file can be installed, `import torch` looks good too. Could I get the commands / config of it?

**Comment by t-vi** [comment]
This sounds external. Note that building PyTorch needs a lot of memory (I'd estimate 2-3GB for MAX_JOBS=1), maybe you can check your dmesg log for the OOM killer.

**Comment by KumaTea** [comment]
> This sounds external. Note that building PyTorch needs a lot of memory (I'd estimate 2-3GB for MAX_JOBS=1), maybe you can check your dmesg log for the OOM killer.

Thank you!

I've also noticed that a few hours ago by monitoring the resources usage,. It turns out the RAM and swap are both full when building this component.

I've assigned another 2GB (since it only has 1GB RAM) for swap, and rebuild it to see what happens. A peak memory usage of about 2.3GB is observed.

Until now everything looks good. If succeed, this issue may be closed. Thanks for your advice again.
